memory/memory/memory.py

```diff
@@ -140,10 +129,23 @@ async def on_startup(app):
     app['client_session'] = httpx.client_session()
     app['worker_pool'] = AsyncWorkerPool(parallelism=100, queue_size=10)
     app['files_in_progress'] = set()
-    app['users'] = {}
+
     kube.config.load_incluster_config()
     k8s_client = kube.client.CoreV1Api()
     app['k8s_client'] = k8s_client
+
+    async def get_secret_from_k8s(secret_name: str) -> GoogleStorageAsyncFS:
+        secret = await retry_transient_errors(
+            k8s_client.read_namespaced_secret, secret_name, DEFAULT_NAMESPACE, _request_timeout=5.0
+        )
+        gsa_key = json.loads(base64.b64decode(secret.data['key.json']).decode())
+        credentials = GoogleCredentials.from_credentials_data(gsa_key)
+        return GoogleStorageAsyncFS(credentials=credentials)
```

Comment:
Is this constructor expensive? If not, it seems cleaner to just use a `K8sCache` here for the gsa key and construct a `GoogleStorageAsyncFS` using the cached secret in `get_or_add_user`.